### ISSUE TITLE
Legacy mode bug on Android 14 with music player apps

### ISSUE THREAD

**Issue by shsui5379** [Pittvandewitt/Wavelet#269]
After updating to Android 14 (Pixel Oct. 2023 stable), when legacy mode is enabled, when you start playback on a music player app (ex: Spotify, Tidal, YouTube Music, etc.), no audible sounds are outputted until you turn Wavelet off then on again.

Specifically affecting playback over USB Audio.

**Comment by acarasimon96** [comment]
Can confirm this myself on Pixel 7a with the latest security updates and Bluetooth audio. I've never had this happen before I updated my system to Android 14 last October.

**Comment by Pittvandewitt** [comment]
Does this issue persist using the update released in the beta test?

**Comment by acarasimon96** [comment]
Yes, the bug still exists in beta v23.11

**Comment by acarasimon96** [comment]
I'd like to add that I can't reproduce this on speaker output if no other audio device is connected to my phone

**Comment by Pittvandewitt** [comment]
This leads me to believe that the file where the settings are stored from and read to is not correctly synced to the settings you see on your screen when switching from an output device, but this doesn't necessarily have to do with legacy mode. I'll have to rewrite a tiny bit of code to make sure this doesn't happen anymore.

**Comment by Pittvandewitt** [comment]
Possible duplicate of https://github.com/Pittvandewitt/Wavelet/issues/251

**Comment by acarasimon96** [comment]
I don't think it has to do with just switching devices. As stated in earlier comments, this is reproducible _during and after_ connecting the audio device.

However, that doesn't rule out the possibility that it's some sort of settings-related bug that only affects Android 14 users who have Legacy Mode enabled.

**Comment by Pittvandewitt** [comment]
Just to make sure, is there no audible sound or are the changes made by Wavelet not applied to the music?

**Comment by acarasimon96** [comment]
No sound only when Wavelet is already turned on and either the following happens:

* An audio device (Bluetooth or USB) is connected while there is already something playing audio
* I start playback while an audio device is connected and nothing has been playing for at least a few seconds

Again, turning off Wavelet and turning it back on while audio is already playing causes the DSP to work again. The same workaround can be achieved with toggling Legacy Mode off and back on, but then I have to do either workaround again the every single time I start playing music again.

**Comment by Pittvandewitt** [comment]
I was able to reproduce the issue only a handful of times. I think legacy mode is showing its age [after 12 years](https://issuetracker.google.com/issues/36936557#comment2). Do you have virtualisation enabled?

**Comment by Pittvandewitt** [comment]
> Virtualization was turned off when I tested for the bug earlier, and I don't see an option for it when Legacy Mode is enabled (which is expected according to the docs, I think).

Reverberation is not available in legacy mode, but virtualisation is. Are you sure virtualisation was off or are you confusing it with reverberation?

**Comment by acarasimon96** [comment]
Oh, my bad. It was on actually, but I just tested Wavelet again with Virtualizer off and I was able to hear sound when I started playback with Legacy Mode turned on.

I swear I left that feature enabled back when I had Android 13 and never encountered the bug in that Android version.
